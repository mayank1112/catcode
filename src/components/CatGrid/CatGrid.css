.grid {
    display: grid;
    grid-template-columns: auto auto auto;
    height: 210px;
    box-sizing: border-box;
}
.gridItem {
    border: 1px solid #66b0ff;
    text-align: center;
    height: 70px;
}

.gridItem img {
    max-height: 70px;
    animation: grow 1s;
}
  
@keyframes grow {
  from {
    transform: scale(.5);
    opacity: .4;
  }
  to {
    transform: scale(1);
    opacity: 1;
  }
}