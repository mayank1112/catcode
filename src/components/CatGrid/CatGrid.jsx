import CatImage from "../shared/CatImage/CatImage";
import './CatGrid.css';

const CatGrid = ({ cats, groupedCats }) => (
    <div className='grid'>
        {cats && cats.length ? cats.map((cat, index) => (
            <div className='gridItem' key={index}>
                {groupedCats[index] && (
                    <CatImage catId={groupedCats[index]} />
                )}
            </div>
        )) : null}
    </div>);

export default CatGrid;
