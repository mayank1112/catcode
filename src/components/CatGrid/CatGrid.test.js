import Enzyme, { shallow } from 'enzyme';
import CatGrid from './CatGrid';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

it('<CatGrid> component should match snapshot', () => {
    const component = shallow(<CatGrid />);
    expect(toJson(component)).toMatchSnapshot();
});
