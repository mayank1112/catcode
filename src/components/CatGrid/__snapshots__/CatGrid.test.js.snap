// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatGrid> component should match snapshot 1`] = `
<div
  className="grid"
/>
`;
